### PR TITLE
RUST-1498 Automatically test FaaS behavior

### PR DESCRIPTION
**Pull request by abr-egn**
RUST-1498

This spins up an aws lambda function using the standard template that tracks and reports some stats.

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -395,12 +394,20 @@ buildvariants:
...
-    patchable: true  # TODO RUST-1684 change to false before merge
+    patchable: false
     run_on:
       - rhel87-small
     tasks:
       - name: search-index-task-group
 
+  - name: aws-lambda
+    display_name: "AWS Lambda"
+    patchable: true
```
Comment (abr-egn) [inline .evergreen/config.yml:405]: ... but this time I will, swearsies.

File: .evergreen/run-aws-lambda-test.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Needed for cargo-lambda
+pip3 install ziglang
```
Comment (kkloberdanz) [inline .evergreen/run-aws-lambda-test.sh:4]: Are we using Zig for its cross compilation toolchain, i.e., `zig cc`?

**Review by kkloberdanz — APPROVED** [review]
Looks good!

File: .evergreen/run-aws-lambda-test.sh (hunk 2 above)
Comment (abr-egn) [reply .evergreen/run-aws-lambda-test.sh:4]: I believe that's what `cargo-lambda` uses it for, yeah.  The important part from my PoV is it refuses to run if it isn't installed :)

**Review by isabelatkinson — APPROVED** [review]

**Pull request merged**